### PR TITLE
Fixes some suicide vest shit

### PR DESCRIPTION
**Pull request by karmaisblackandbluepilled**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

## Changelog
:cl:
balance: You can't trigger the suicide vest when inside your backpack anymore, wear it, retard.
fix: Made the suicide vest less shitcode?
/:cl:

## About The Pull Request
yoyo is beating me, help

### CODE REVIEW

File: code/modules/clothing/suits/armor.dm [hunk 1]
```diff
@@ -180,13 +172,14 @@
...
 	/obj/item/clothing/suit/armor/suicide/AltClick(mob/living/user)
-		if(attached_grenade.len != 0)
-			playsound(src, "allah", 100, 0)
-			for(var/obj/item/grenade/G in attached_grenade)
-				addtimer(CALLBACK(G, /obj/item/grenade.proc/prime), 1)
-			sleep(3)
-			qdel(src)
-
+		if(istype(user.get_item_by_slot(SLOT_WEAR_SUIT), /obj/item/clothing/suit/armor/suicide))
+			if(attached_grenade.len != 0)
```
Comment (yoyobatty) [inline code/modules/clothing/suits/armor.dm]: this should just be `if(attached_grenade.len)`

File: code/modules/clothing/suits/armor.dm [hunk 2]
```diff
@@ -168,7 +161,6 @@
 				G.forceMove(src)
 		else
 			return ..()
-
 	/obj/item/clothing/suit/armor/suicide/attack_self(var/mob/user)
 		if(length(attached_grenade) != 0)
```
Comment (yoyobatty) [inline code/modules/clothing/suits/armor.dm]: should just be `if(attached_grenade.len)`

File: code/modules/clothing/suits/armor.dm [hunk 3]
```diff
@@ -180,13 +172,14 @@
...
-			qdel(src)
-
+		if(istype(user.get_item_by_slot(SLOT_WEAR_SUIT), /obj/item/clothing/suit/armor/suicide))
+			if(attached_grenade.len != 0)
+				playsound(src, "allah", 100, 0)
+				for(var/obj/item/grenade/G in attached_grenade)
+					addtimer(CALLBACK(G, /obj/item/grenade.proc/prime), 1)
+				QDEL_IN(src, 3)
+		else
+			to_chat(user, "You need to be wearing the suit to trigger it.")
```
Comment (yoyobatty) [inline code/modules/clothing/suits/armor.dm:181]: put `return` after the `to_chat` line just incase

**Comment by yoyobatty** [comment]
Tested and works?

**Comment by karmaisblackandbluepilled** [comment]
i didn't test this yet, why'd you merge it
did you test it?

**Comment by yoyobatty** [comment]
It's not merged at all

**Comment by karmaisblackandbluepilled** [comment]
github is a mystery

**Comment by karmaisblackandbluepilled** [comment]
ok i just tested it, it works
you can't trigger it while not wore and the rest of the shit works too

**Comment by MrEpicSax** [comment]
+1 if this is a fix why you need votes?

**Comment by AnthonyJNgo** [comment]
It's not a fix it's a tweak/slight change. A fix would be patching a bug. With that said +3

**Pull request merged**